src/pybind/mgr/volumes/fs/stats_util.py

```diff
@@ -96,27 +100,93 @@ def get_num_ratio_str(num1, num2, human=True):
     return f'{num1}/{num2}'
 
 
-def get_stats(src_path, dst_path, fsh, human=True, should_cancel=False):
+def get_amount_copied(src_path, dst_path, fsh, human=True):
+    rbytes = 'ceph.dir.rbytes'
+
+    size_t = int(fsh.getxattr(src_path, rbytes))
+    size_c = int(fsh.getxattr(dst_path, rbytes))
+
+    if size_t == 0:
+        return -1 if size_c == 0 else -2
+
+    percent = ((size_c/size_t) * 100)
+    percent = round(percent, 3)
+
+    return size_t, size_c, percent
+
+
+def get_stats(src_path, dst_path, fsh, human=True):
     rfiles = 'ceph.dir.rfiles'
     rsubdirs = 'ceph.dir.rsubdirs'
-    rbytes = 'ceph.dir.rbytes'
 
     rfiles_t = int(fsh.getxattr(src_path, rfiles))
     subdirs_t = int(fsh.getxattr(src_path, rsubdirs))
-    size_t = int(fsh.getxattr(src_path, rbytes))
 
     rfiles_c = int(fsh.getxattr(dst_path, rfiles))
     subdirs_c = int(fsh.getxattr(dst_path,rsubdirs))
-    size_c = int(fsh.getxattr(dst_path, rbytes))
 
-    if size_t == 0:
-        return -1 if size_c == 0 else -2
-    percent = ((size_c/size_t) * 100)
-    percent = round(percent, 2)
+    size_t, size_c, percent = get_amount_copied(src_path, dst_path, fsh,
+                                                human)
 
     return {
         'percentage cloned': percent,
         'amount cloned': get_size_ratio_str(size_c, size_t, human),
         'regfiles cloned': get_num_ratio_str(rfiles_c, rfiles_t, human),
         'subdirs cloned': get_num_ratio_str(subdirs_c, subdirs_t, human)
     }
+
+
+class CloneProgressBar:
+
+    def __init__(self, vc, volname, dst_svname, src_path, dst_path):
+        self.vc = vc
+        self.volname = volname
+        self.dst_svname = dst_svname
+
+        self.src_path = src_path
+        self.dst_path = dst_path
+
+        self.pev_id = str(uuid4())
```

Comment:
No this is not unique for each clone, this line will be executed just once.

Reply:
That's because in one of the later commit one single object was instantiated (in one of the previous commits, there used to be one object per clone), correct?

Reply:
Right.